examples/p4est_2d_dgsem/elixir_euler_supersonic_cylinder_amr.jl

```diff
@@ -0,0 +1,139 @@
+                                                      surface_flux_function, equations::CompressibleEulerEquations2D)
+  u_boundary = initial_condition_mach3_flow(x, t, equations)
+  flux = Trixi.flux(u_boundary, normal_direction, equations)
+
+  return flux
+end
+
+
+# Supersonic outflow boundary condition.
+# Calculate the boundary flux entirely from the internal solution state. Analogous to supersonic inflow
+# except all the solution state values are set from the internal solution as everything leaves the domain
+@inline function boundary_condition_outflow(u_inner, normal_direction::AbstractVector, x, t,
+                                            surface_flux_function, equations::CompressibleEulerEquations2D)
+  flux = Trixi.flux(u_inner, normal_direction, equations)
+
+  return flux
+end
+
+boundary_conditions = Dict( :Bottom  => boundary_condition_slip_wall,
+                            :Circle  => boundary_condition_slip_wall,
+                            :Top     => boundary_condition_slip_wall,
+                            :Right   => boundary_condition_outflow,
+                            :Left    => boundary_condition_supersonic_inflow )
+
+volume_flux = flux_ranocha_turbo
+surface_flux = flux_lax_friedrichs
+
+polydeg = 3
+basis = LobattoLegendreBasis(polydeg)
+shock_indicator = IndicatorHennemannGassner(equations, basis,
+                                            alpha_max=0.5,
+                                            alpha_min=0.001,
+                                            alpha_smooth=true,
+                                            variable=density_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(shock_indicator;
+                                                 volume_flux_dg=volume_flux,
+                                                 volume_flux_fv=surface_flux)
+solver = DGSEM(polydeg=polydeg, surface_flux=surface_flux, volume_integral=volume_integral)
+
+# Get the unstructured quad mesh from a file (downloads the file if not available locally)
+default_mesh_file = joinpath(@__DIR__, "abaqus_cylinder_in_channel.inp")
+isfile(default_mesh_file) || download("https://gist.githubusercontent.com/andrewwinters5000/a08f78f6b185b63c3baeff911a63f628/raw/addac716ea0541f588b9d2bd3f92f643eb27b88f/abaqus_cylinder_in_channel.inp",
+                                      default_mesh_file)
+mesh_file = default_mesh_file
+
+mesh = P4estMesh{2}(mesh_file)
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
+                                    boundary_conditions=boundary_conditions)
+
+###############################################################################
+# ODE solvers with workaround to catch location of any negative density and/or pressure
+
+tspan = (0.0, 2.0)
+ode = semidiscretize(semi, tspan)
+
+###############################################################################
+# Callbacks
```

Comment:
```suggestion
# ODE solvers

tspan = (0.0, 2.0)
ode = semidiscretize(semi, tspan)

# Callbacks
```
I don't see the workaround anymore?

Reply:
Right I took it out and have a different elixir locally with this debugging feature. I just forgot to update the comments.